internal/pkg/pkg_template/package_template.go

```diff
@@ -0,0 +1,59 @@
+package pkg_template
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+)
+
+var tarUtil tarHelpers
+var gzipUtil gzipHelpers
+var ioUtil ioUtils
+
+func init() {
+	tarUtil = tarHelpersImpl{}
+	ioUtil = ioUtilHelpersImpl{}
+	gzipUtil = gzipHelpersImpl{}
+}
+
+func PackageTemplate(templatePath string) (archiveFilePath string, err error) {
```

Comment:
I don't see the need for this to be a separate package. This is directly relevant to the `pct` package in that it builds and packages templates, so it should be over there and not in a new package. As a bonus you won't have to duplicate test files and other things. 

The gzip and tar can be their own packages, as they are outside concerns to the main goal.

In summary, create a new file called `pct/build.go` and put `package_template.go` content there.